mvpa2/datasets/mri.py

```diff
@@ -270,8 +283,18 @@ def fmri_dataset(samples, targets=None, chunks=None, mask=None,
             ds.fa[fattr] = ds.a.mapper.forward1(value)
 
     # store interesting props in the dataset
-    ds.a['imghdr'] = imghdr
-    ds.a['imgtype'] = imgtype
+    ds.a['imgaffine'] = img.get_affine()
```

Comment:
get_best_affine() is a header function. Images only have get_affine(). It should do the right thing.
FYI: Not all image headers have affines. In the MINC case the image is the only object that has it.